x-pack/plugins/triggers_actions_ui/public/application/lib/check_action_type_enabled.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiCard, EuiLink } from '@elastic/eui';
+import { ActionType } from '../../types';
+import { VIEW_LICENSE_OPTIONS_LINK } from '../../common/constants';
+import './check_action_type_enabled.scss';
+
+export interface IsEnabledResult {
+  isEnabled: true;
+}
+export interface IsDisabledResult {
+  isEnabled: false;
+  message: string;
+  messageCard: JSX.Element;
+}
+
+export function checkActionTypeEnabled(
+  actionType?: ActionType
+): IsEnabledResult | IsDisabledResult {
+  if (actionType?.enabledInLicense === false) {
+    return {
+      isEnabled: false,
+      message: i18n.translate(
+        'xpack.triggersActionsUI.checkActionTypeEnabled.actionTypeDisabledByLicenseMessage',
+        {
+          defaultMessage:
+            'This connector is disabled because it requires a {minimumLicenseRequired} license.',
+          values: {
+            minimumLicenseRequired: actionType.minimumLicenseRequired,
+          },
+        }
+      ),
+      messageCard: (
+        <EuiCard
+          title={i18n.translate(
+            'xpack.triggersActionsUI.sections.alertForm.actionTypeDisabledByLicenseMessageTitle',
+            {
+              defaultMessage: 'This feature requires a {minimumLicenseRequired} license.',
+              values: {
+                minimumLicenseRequired: actionType.minimumLicenseRequired,
+              },
+            }
+          )}
+          description={i18n.translate(
+            'xpack.triggersActionsUI.sections.alertForm.actionTypeDisabledByLicenseMessageDescription',
+            { defaultMessage: 'To re-enable this action, please upgrade your license.' }
```

Comment:
Ya, I check this condition, and it does do the right thing when an action becomes not enabled (revert from trial to basic).  Wasn't clear from the flow though.  I'm guessing there's never a state where this could be displayed if it already wasn't enabled previously.   Probably worth a comment, if it is the case that currently that's the only way to get here.

Reply:
Yeah, I was thinking at first to just remove the `re-` portion but then realized this will get a review from UI copy. So I went ahead and just put a comment instead: https://github.com/elastic/kibana/pull/59819/commits/d8c6b4743a40f9c9635911c11af779714e555d42 :)